### PR TITLE
Support pfx certificates

### LINKED ISSUE #359: SSL certificates in pfx format
Working in a windows based environment, I would like to use my pfx certificate which I use in IIS also with verdaccio, but a quick glimpse in the code showed me, that this is not supported ...

As node also supports pfx (even with a certificate chain) it should be quite easy to support this also in verdaccio

### PR DESCRIPTION
**Pull request by wiggisser**
**Type:**  feature

The following has been addressed in the PR:

*  There is a related issue

**Description:**

Adds support to use a certificate in pfx file format with an optional passphrase

Resolves #359

### CODE REVIEW

File: src/lib/cli.js [hunk 1]
```diff
@@ -143,13 +146,24 @@ function afterConfigLoad() {
       }
 
       try {
-        webServer = https.createServer({
+        let httpsoptions = {
```
[the remark this answers is not in the document]
Comment (juanpicado) [reply src/lib/cli.js]: `httpsoptions` won't change, so, I'd make sense use `const` instead.

**Comment by wiggisser** [comment]
I did the requested changes, but I don't think I can do anything about the failing test?

**Comment by juanpicado** [comment]
Don't worry, it was an issue with `circleci`. Fixed.

**Comment by juanpicado** [comment]
One more thing, can you amend the commit messages following https://seesparkbox.com/foundry/semantic_commit_messages ? 

Since this is a new feature. It would be: 

`feat: add pfx support for https`
`refactor: use object assign`
`chore: fix ci errors`
`refactor: camelCasing and const binding`

I don't want to squash your commits and also that helps a lot to generate automatically the Changelong.

**Comment by juanpicado** [comment]
Great, thanks guys !  Next minor version is yours 🍸

**Comment by juanpicado** [comment]
@wiggisser we forgot update the documentation https://github.com/verdaccio/verdaccio/blob/master/wiki/ssl.md

**Comment by wiggisser** [comment]
My pleasure. Thanks for accepting my PR that quickly. Created a separate PR for the docs ...

**Pull request merged**